Marvell prestera kernel driver

antony-rheneus opened the pull request:
Build Marvell kernel driver for prestera sai sdk
Builds interrupt and dma kernel driver as part of sonic build

mvIntDrv.ko and mvDmaDrv.ko modules are loaded using etc/modules-load.d/marvell.conf script 

<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it

#### How I did it

#### How to verify it

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


#### A picture of a cute animal (not mandatory but encouraged)

lguohan commented:
can you remove replaced kernel driver?

lguohan commented on platform/marvell-armhf/prestera/debian/rules:
```diff
@@ -0,0 +1,53 @@
+build:
+	# get sources
+	rm -rf mrvl-prestera || true
+	git clone https://github.com/Marvell-switching/mrvl-prestera.git
```
we need tag to checkout a specific commit.

otherwise, the build is not reproducible.

antony-rheneus replied:
Ok sure, I will add git tag, and update here accordingly

antony-rheneus replied:
Addressed your comment, by adding git tag, and updated the git clone url with the TAG.

antony-rheneus commented:
> can you remove replaced kernel driver binary?

Ok sure, I will remove it, and add as part of this PR too.

antony-rheneus commented:
> > can you remove replaced kernel driver binary?
> 
> Ok sure, I will remove it, and add as part of this PR too.

Done, removed the kernel binary  package by removing all its makefile content
Now it is replaced with the prestera.mk which will download the source and build inside sonic buildimage.

antony-rheneus commented:
retest this please

antony-rheneus commented:
retest this please

antony-rheneus commented:
@lguohan , I hav addressed your comments, on maintaining git tag for the driver source, as well removing old ko binaries from the build.
Would you please review and confirm?

lguohan commented on platform/marvell-armhf/rules.mk:
```diff
@@ -5,9 +5,9 @@ include $(PLATFORM_PATH)/docker-syncd-mrvl-rpc.mk
 include $(PLATFORM_PATH)/docker-saiserver-mrvl.mk
 include $(PLATFORM_PATH)/libsaithrift-dev.mk
 include $(PLATFORM_PATH)/one-image.mk
-include $(PLATFORM_PATH)/linux-kernel-armhf.mk
```
can you remove this file if it is not used.

antony-rheneus replied:
Thanks for pointing it out, I have removed the obsolete files now.

lguohan commented:
can we get this one to 202012 branch?

antony-rheneus commented:
> can we get this one to 202012 branch?

Ok sure, I will port it to 202012 branch.
